kubeflow/trainer/options/options.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2025 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Optional, Protocol
+
+
+class Option(Protocol):
+    """Protocol for TrainJob configuration options.
+
+    Options provide a composable way to configure different aspects of a TrainJob.
+    Each option implements the __call__ method to modify the TrainJob specification.
+    """
+
+    def __call__(self, job_spec: dict) -> None:
+        """Apply this option to the TrainJob specification.
+
+        Args:
+            job_spec: The TrainJob specification dictionary to modify.
+        """
+        ...
+
+
+@dataclass
+class WithLabels:
+    """Add labels to the TrainJob resource metadata (.metadata.labels).
+
+    These labels are applied to the TrainJob resource itself and are used
+    for resource organization, filtering, and selection.
+
+    Args:
+        labels: Dictionary of labels to apply to the TrainJob metadata.
+    """
+
+    labels: dict[str, str]
+
+    def __call__(self, job_spec: dict) -> None:
+        """Apply labels to TrainJob metadata."""
+        metadata = job_spec.setdefault("metadata", {})
+        existing_labels = metadata.setdefault("labels", {})
+        existing_labels.update(self.labels)
```

Comment:
Shall we override labels here since `WithLabels()` will always set the actual labels for the TrainJob ?